src/manage_nsfs/diagnose.js

```diff
@@ -0,0 +1,64 @@
+/* Copyright (C) 2024 NooBaa */
+'use strict';
+
+const dbg = require('../util/debug_module')(__filename);
+const health = require('./health');
+const config = require('../../config');
+const os_util = require('../util/os_utils');
+const { DIAGNOSE_ACTIONS } = require('./manage_nsfs_constants');
+const ManageCLIError = require('./manage_nsfs_cli_errors').ManageCLIError;
+const { throw_cli_error, write_stdout_response } = require('./manage_nsfs_cli_utils');
+const ManageCLIResponse = require('../manage_nsfs/manage_nsfs_cli_responses').ManageCLIResponse;
+
+/**
+ * manage_diagnose_operations handles cli diagnose operations
+ * @param {string} action 
+ * @param {Object} user_input 
+ * @param {Object} global_config 
+ * @returns {Promise<Void>}
+ */
+async function manage_diagnose_operations(action, user_input, global_config) {
+    switch (action) {
+        case DIAGNOSE_ACTIONS.HEALTH:
+            await health.get_health_status(user_input, global_config);
+            break;
+        case DIAGNOSE_ACTIONS.GATHER_LOGS:
+            await gather_logs();
+            break;
+        case DIAGNOSE_ACTIONS.METRICS:
+            await gather_metrics();
+            break;
+        default:
+            throw_cli_error(ManageCLIError.InvalidAction);
+    }
+}
+
+/**
+ * gather_logs handles cli diagnose gather-logs operation
+ * @returns {Promise<Void>}
+ */
+async function gather_logs() {
+    throw_cli_error(ManageCLIError.NotImplemented);
+}
+
+/**
+ * gather_metrics handles cli diagnose metrics operation
+ * @returns {Promise<Void>}
+ */
+async function gather_metrics() {
+    const metrics_query = `curl -s http://127.0.0.1:${config.EP_METRICS_SERVER_PORT}/metrics/nsfs_stats`;
```

Comment:
I think relying on external curl on the host is not needed and we should use http.request where possible.

we should add a function like this in http_utils (like the make_https_request we already have):
```js
async function make_http_request(options, body) {
    return new Promise((resolve, reject) => {
        http.request(options, resolve)
            .on('error', reject)
            .end(body);
    });
}
```

and then can call -
```js
const res = await http_utils.make_http_request({
    hostname: 'localhost',
    port: config.EP_METRICS_SERVER_PORT,
    path: '/metrics/nsfs_stats',
});
const metrics_output = (await buffer_utils.read_stream_join(res)).toString();

```